### PR TITLE
Improve favicons to add support for different browsers and use cases

### PR DESCRIPTION
**Pull request by fichtnerma**
---
Improve favicons to add support for different browsers and use cases

## PR Checklist

-   [x] Link to the respective task if one exists: COM-1012
-   [x] Provide screenshots/screencasts if the change contains visual changes

### CODE REVIEW

File: site/src/app/manifest.webmanifest [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "icons": [
+        { "src": "/icon-192.png", "type": "image/png", "sizes": "192x192" },
+        { "src": "/icon-512.png", "type": "image/png", "sizes": "512x512" }
```
Comment (johnnyomair) [inline site/src/app/manifest.webmanifest:4]: I just tested this locally, both these icons can't be found. Please make sure that this works.

**Review by johnnyomair — CHANGES_REQUESTED** [review]

File: site/src/app/manifest.webmanifest (hunk 1 above)
Comment (fichtnerma) [reply site/src/app/manifest.webmanifest:4]: It should work now

**Comment by johnnyomair** [comment]
I just had a look at the article this PR is based on: https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs. Some parts are only required when making a PWA. Do we still want to support the PWA parts?

**Comment by nsams** [comment]
> Some parts are only required when making a PWA

I don't know what a PWA exactly is, but having PWA compatible favicons - and nothing else - doesn't make it one. I'd keep it simple and only do things that actually solve problems.

Maybe the ones supported by [nextjs](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#image-files-ico-jpg-png) are enough.

**Comment by johnnyomair** [comment]
> Maybe the ones supported by [nextjs](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#image-files-ico-jpg-png) are enough.

Sounds good. @fichtnerma could you please remove the manifest and the icons in public/? Thanks!

File: site/src/middleware.ts [hunk 2]
```diff
@@ -97,12 +97,12 @@ export const config = {
          * - api (API routes)
          * - _next/static (static files)
          * - _next/image (image optimization files)
-         * - favicon.ico, favicon.svg, favicon.png
-         * - manifest.json
+         * - favicon.ico, icon.svg, favicon.png, icon-192.png, icon-512.png, apple-icon.png
+         * - manifest.json, manifest.webmanifest
```
Comment (johnnyomair) [inline site/src/middleware.ts:101]:
```suggestion
         * - favicon.ico, icon.svg, apple-icon.png
         * - manifest.json
```

**Review by johnnyomair — APPROVED** [review]

**Comment by nsams** [comment]
apple-icon is still needed even if there is a svg version of icon?

**Comment by fichtnerma** [comment]
> apple-icon is still needed even if there is a svg version of icon?

Yes the apple-touch-icon is needed on iOS when placing the website on the Homescreen

**Comment by nsams** [comment]
> Yes the apple-touch-icon is needed on iOS when placing the website on the Homescreen

and iOS really doesn't use the `<link rel="icon"` if there is no `<link rel="apple-touch-icon"`?

**Comment by fichtnerma** [comment]
> > Yes the apple-touch-icon is needed on iOS when placing the website on the Homescreen
> 
> and iOS really doesn't use the `<link rel="icon"` if there is no `<link rel="apple-touch-icon"`?

Yeah without that icon there will just be no icon displayed

**Review by nsams — APPROVED** [review]

**Pull request merged**